plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py

```diff
@@ -44,7 +44,9 @@ def get_package_xml(pkg_path, sumtype=util.TYPE_SHA256):
         return {}
     # RHEL6 createrepo throws a ValueError if _cachedir is not set
     po._cachedir = None
-    primary_xml_snippet = change_location_tag(po.xml_dump_primary_metadata(), pkg_path)
+    primary_xml_snippet = po.xml_dump_primary_metadata()
+    primary_xml_snippet = primary_xml_snippet.decode('utf-8', 'replace')
```

Comment:
I notice that you moved this to be sooner in the workflow than where ``string_to_unicode`` was being called. What's the reasoning for that?